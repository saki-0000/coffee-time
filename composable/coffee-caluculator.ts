import { ref } from '@nuxtjs/composition-api'

export const useCoffeeCaluculator = () => {
  const coffee = ref(0)
  const water = ref(0)
  const onInputWater = () => {
    coffee.value = Math.round(water.value * (7 / 100) * 10) / 10
  }
  const onInputCoffee = () => {
    water.value = Math.round(coffee.value * (100 / 7) * 10) / 10
  }
  return {
    coffee,
    water,
    onInputWater,
    onInputCoffee,
  }
}
