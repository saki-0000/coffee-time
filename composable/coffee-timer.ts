export const useCoffeeTimer = () => {
  const getLabel = (time: number) => {
    if (time === 180) {
      return 'コーヒーを淹れる'
    } else if (time < 180 && time > 120) {
      return '1湯目...'
    } else if (time <= 120 && time > 60) {
      return '2湯目...'
    } else if (time <= 60 && time > 0) {
      return '3湯目...'
    } else if (time === 0) {
      return 'Have a Good Time!'
    }
  }
  return {
    getLabel,
  }
}
