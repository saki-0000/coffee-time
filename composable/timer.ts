import { ref, computed } from '@nuxtjs/composition-api'
import { format } from 'date-fns'

export const useTimer = (initTime = 180) => {
  const _time = ref(180)
  const countDown = () => _time.value--
  const measureTime = () => {
    const intervalId = setInterval(() => {
      countDown()
      if (time.value === 0) {
        clearInterval(intervalId)
      }
    }, 1000)
  }
  const time = computed(() => _time.value)
  const formatTime = (time: number) => {
    return format(new Date(0, 0, 0, 0, 0, time), 'mm:ss')
  }
  return {
    measureTime,
    time,
    formatTime,
  }
}
