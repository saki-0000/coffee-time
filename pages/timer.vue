<template>
  <v-row justify="center" align="center">
    <v-col cols="12" sm="8" md="6">
      <v-card class="logo py-4 d-flex justify-center">
        <v-text-field
          v-model.number="second"
          type="number"
          label="秒"
          hint="何秒はかりますか？？"
        />
        <Timer :second="second" />
      </v-card>
    </v-col>
  </v-row>
</template>

<script lang="ts">
import { defineComponent } from '@vue/composition-api'
import Timer from '../components/Timer.vue'
export default defineComponent({
  name: 'TimerPage',
  components: { Timer },
  setup: () => {
    const second: Number = 180
    return {
      second,
    }
  },
})
</script>
