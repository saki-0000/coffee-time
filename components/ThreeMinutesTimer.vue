<template>
  <div>
    <v-btn id="startButton" :disabled="disabled" @click="onClick"
      ><slot>スタート</slot></v-btn
    >
    <span id="displayTime">
      {{ formatTime(time) }}
    </span>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, watchEffect } from '@nuxtjs/composition-api'
import { useTimer } from '~/composable/timer'

export default defineComponent({
  props: {
    second: {
      type: Number,
      default: 180,
    },
  },
  emits: ['change'],
  setup: (_, context) => {
    const { measureTime, time, formatTime } = useTimer()
    const disabled = ref(false)
    watchEffect(() => {
      context.emit('change', time.value)
    })
    const onClick = () => {
      disabled.value = true
      measureTime()
    }
    return {
      disabled,
      time,
      onClick,
      formatTime,
    }
  },
})
</script>
