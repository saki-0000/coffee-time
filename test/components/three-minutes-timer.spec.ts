import { shallowMount, Wrapper } from '@vue/test-utils'
import ThreeMinutesTimer from '~/components/ThreeMinutesTimer.vue'
import * as composable from '~/composable/timer'

describe('ThreeMinutesTimer.vue', () => {
  let wrapper: Wrapper<any>
  beforeEach(() => {
    wrapper = shallowMount(ThreeMinutesTimer)
  })

  test('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('call useTime`s functions when the `start` button clicked ', () => {
    wrapper.get('#startButton').trigger('click')
    expect(wrapper.get('#displayTime').text()).toBe('03:00')
  })
  test('v-btn to be disabled when the `start` button clicked ', () => {
    wrapper.get('#startButton').trigger('click')
    expect(wrapper.get('#displayTime').text()).toBe('03:00')
  })
})
