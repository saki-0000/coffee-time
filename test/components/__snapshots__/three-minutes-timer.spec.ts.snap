// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreeMinutesTimer.vue renders correctly 1`] = `
"<div>
  <v-btn-stub tag=\\"button\\" activeclass=\\"\\" type=\\"button\\" id=\\"startButton\\">スタート</v-btn-stub> <span id=\\"displayTime\\">
    100
  </span>
</div>"
`;
